### PR TITLE
Ch 774 MVP A/B reports page

### PR DESCRIPTION
**Pull request by eshtadc**

### CODE REVIEW

File: acquia_lift.admin.inc [hunk 1]
```diff
@@ -261,6 +261,27 @@ function acquia_lift_report($form, &$form_state, $agent_data, $option_set) {
     );
   }
 
+  // @todo change this to the interface once simplified campaign creation code
+  // has been merged into the main branch.
```
Comment (katbailey) [inline acquia_lift.admin.inc]: Is this todo still relevant?

Comment (eshtadc) [inline acquia_lift.admin.inc]: Nope :-)

File: includes/acquia_lift.classes.inc [hunk 2]
```diff
@@ -1282,23 +1293,23 @@ interface AcquiaLiftReportDataSourceInterface {
...
+   *   are the parameter names and the values are the parameter values.
+   *   Support includes:
+   *     - features: An array of features to include in the report.
+   *       Defaults to "none" but can also be set to "all" for all features.
+   *     - confidence-measure: The confidence measure to use between 0 and 1.
+   *     - aggregated-over-dates: Aggregates the data over the time span or
+   *       reports on individual dates
    * @return array
    *   The report as an associative array with the structure as defined at
    *   https://console.lift.acquia.com/docs/abc/reporting-api under "Option
```
Comment (katbailey) [inline includes/acquia_lift.classes.inc]: This is a good opportunity to remove this reference to a non-existent API doc URL :)

File: js/acquia_lift.reports.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+/**
+ * Functionality related specifically to the modal campaign management
+ * procedures.
```
Comment (katbailey) [inline js/acquia_lift.reports.js]: orly? (Why yes, I did just scroll through ~1000 lines of js with nothing to say except this :P)

File: acquia_lift.module [hunk 4]
```diff
@@ -1385,6 +1417,114 @@ function acquia_lift_libraries_info() {
...
+    'variant order' => array('minified', 'source'),
+    'version arguments' => array(
+      'variants' => array(
+        'source' => array(
+          'file' => 'rickshaw.js',
+          'pattern' => '/(Rickshaw)/',
+        ),
+        'minified' => array(
+          'file' => 'rickshaw.min.js',
+          'pattern' => '/(Rickshow)/',
```
Comment (katbailey) [inline acquia_lift.module]: "Rickshow" - for once my inability to get the library to show up was not my own fault! :P

File: acquia_lift.module [hunk 5]
```diff
@@ -1385,6 +1417,114 @@ function acquia_lift_libraries_info() {
...
+            // Without a variant callback, the variant is assumed to not be
+            // installed.
+            'variant callback' => '_acquia_lift_libraries_variant_exists',
+            'variant arguments' => array('d3.min.js'),
+          ),
+        ),
+      ),
+    ),
+  );
+
```
Comment (katbailey) [inline acquia_lift.module:1527]: The acquia_lift.make.example file needs to be updated to include these new libraries

File: acquia_lift.module [hunk 6]
```diff
@@ -1530,6 +1670,54 @@ function _acquia_lift_libraries_get_preferred_variant_name($library) {
...
+    }
+    drupal_set_message(theme('item_list', array(
+      'type' => 'ul',
+      'title' => $missing_message,
+      'items' => $items,
+    )), 'error');
+    return TRUE;
+  }
+  return FALSE;
+}
```
Comment (katbailey) [inline acquia_lift.module:1718]: This was very helpful :)

File: plugins/agent_types/AcquiaLiftAgent.inc [hunk 7]
```diff
@@ -1018,6 +1014,613 @@ class AcquiaLiftReport implements PersonalizeAgentReportInterface, AcquiaLiftRep
...
 
   /**
+   * Generates the variation abbreviated label.
+   *
+   * @param $counter
+   *   Indicates the number for the variation.
+   * @param $is_control
+   *   True if this is the control option.
+   */
+  function getVariationLabel($counter, $is_control) {
```
Comment (katbailey) [inline plugins/agent_types/AcquiaLiftAgent.inc]: Should be marked protected

**Pull request closed without merging**